testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/util/URLAssert.java

```diff
@@ -87,31 +92,33 @@ public static void assertCurrentUrlStartsWith(final AbstractPage page) {
    }
 
     public static void assertCurrentUrlStartsWith(final String url){
-        assertTrue("URL expected to begin with: " + removeDefaultPorts(url) + " ; actual URL: " + DroneUtils.getCurrentDriver().getCurrentUrl(),
-        currentUrlStartsWith(removeDefaultPorts(url)));
-    }
-
-    public static void waitUntilUrlStartsWith(String url, int timeOutInSeconds) {
-        new WebDriverWait(DroneUtils.getCurrentDriver(), timeOutInSeconds).until(ExpectedConditions.urlMatches("^" + url));
+        awaitUntilAsserted(() -> {
+            assertTrue("URL expected to begin with: " + removeDefaultPorts(url) + " ; actual URL: " + DroneUtils.getCurrentDriver().getCurrentUrl(),
+                    currentUrlStartsWith(removeDefaultPorts(url)));
+        });
     }
 
     public static void assertCurrentUrlDoesntStartWith(final AbstractPage page, WebDriver driver) {
         assertCurrentUrlDoesntStartWith(page.toString(), driver);
     }
 
     public static void assertCurrentUrlDoesntStartWith(final String url, WebDriver driver) {
-        DroneUtils.addWebDriver(driver);
-        assertCurrentUrlDoesntStartWith(url);
-        DroneUtils.removeWebDriver();
+        awaitUntilAsserted(() -> {
+            DroneUtils.addWebDriver(driver);
```

Comment:
I think this can be outside of `awaitUntilAsserted`

Reply:
Thanks for the review, updated the code. CI is now running.